File: 1249-minimum-remove-to-make-valid-parentheses/1249-minimum-remove-to-make-valid-parentheses.py
class Solution:
    def minRemoveToMakeValid(self, s: str) -> str:
        
        stack = []
        res = []

        for char in range(len(s)):
            if s[char].isalpha():
                res.append(s[char])
            elif s[char] == "(":
                stack.append(char)
                res.append(s[char])
            elif s[char] == ")":
                if stack and s[stack[-1]] == "(":
                    stack.pop()
                    res.append(s[char])
                else:
                    stack.append(char)
                    res.append(s[char])

        while stack:
            temp = stack.pop()
            del res[temp]

        return "".join(res)

                    

File: 1249-minimum-remove-to-make-valid-parentheses/1249-minimum-remove-to-make-valid-parentheses.cpp
class Solution {
public:
    string minRemoveToMakeValid(string s) 
    {

        std::string res = "";
        std::stack<int> x;

        for (char c : s)
        {
            if (std::isalnum(c))
            {
                res.push_back(c);
            }
            else if (c == '(')
            {
                x.push(res.size());
                res.push_back(c);
            }
            else if (c == ')')
            {
                if (!x.empty())
                {
                    x.pop();
                    res.push_back(c);
                }
                else
                {

                }
            }
        }   

        //Remove extra shit
        while (!x.empty()) 
        {
            res.erase(res.begin() + x.top());
            x.pop();
        }

        return res; 
    }
};